packages/open-payments/src/client/incoming-payment.ts

```diff
@@ -137,6 +165,48 @@ export const completeIncomingPayment = async (
   }
 }
 
+export const listIncomingPayment = async (
+  deps: BaseDeps,
+  args: ListGetArgs,
+  validateOpenApiResponse: ResponseValidator<IncomingPaymentPaginationResult>,
+  pagination?: PaginationArgs
+) => {
+  const { axiosInstance, logger } = deps
+  const { accessToken, paymentPointer } = args
+
+  const url = `${paymentPointer}${getRSPath('/incoming-payments')}`
+
+  const incomingPayments = await get(
+    { axiosInstance, logger },
+    {
+      url,
+      accessToken,
+      ...(pagination ? { queryParams: { ...pagination } } : {})
+    },
+    validateOpenApiResponse
+  )
+
+  for (const incomingPayment of incomingPayments.result) {
+    try {
+      validateIncomingPayment(incomingPayment)
```

Comment:
For list responses, we _could_ validate that `pagination.hasPreviousPage` is `false` if `cursor` was not specified (which should be forward pagination from the start).

Reply:
Mostly agree, but I'm not sure it's worth failing the whole request over it, however.